src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,13 +8,23 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "311, Clementi Ave 2, #02-25, 182938";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format: BLOCK, STREET, UNIT NUMBER, POSTAL CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+,.+,.+,.+";
 
-    public final String value;
+    public String value;
     private boolean isPrivate;
-
+	
+	private Block block;
+	private Street street;
+	private UnitNumber unitNumber;
+	private PostalCode postalCode;
+	
+	private final int BLOCK_INDEX = 0;
+	private final int STREET_INDEX = 1;
+	private final int UNIT_NUMBER_INDEX = 2;
+	private final int POSTAL_CODE_INDEX = 3;
```

Comment:
Good job for using constant to remove magic numbers. For this particular case, you can also check `enum`.